src/features/DebugSession.ts

```diff
@@ -72,23 +80,25 @@ export class DebugSessionFeature extends LanguageClientConsumer
         context.subscriptions.push(vscode.debug.registerDebugAdapterDescriptorFactory("PowerShell", this));
     }
 
-    createDebugAdapterDescriptor(
-        session: vscode.DebugSession,
-        _executable: vscode.DebugAdapterExecutable | undefined): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
+    // This is our factory entrypoint hook to when a debug session starts, and where we will lazy initialize everything needed to do the debugging such as a temporary console if required
+    public async createDebugAdapterDescriptor(
+        session: DebugSession,
+        _executable: DebugAdapterExecutable | undefined): Promise<DebugAdapterDescriptor | undefined> {
+        // NOTE: A Promise meets the shape of a ProviderResult, which allows us to make this method async.
```

Comment:
NICE.

Reply:
Yeah I felt pretty clever when I figured that out.